src/lib/swim/swim.c

```diff
@@ -544,8 +544,22 @@ struct swim {
 	 * lines.
 	 */
 	struct swim_task round_step_task;
+	/** Trigger for fiber shutdown event. */
+	struct trigger on_fiber_shutdown;
 };
 
+/**
+ * Cancel and wait finish of an event handler fiber. That
+ * operation is not inlined in the SWIM destructor, because there
+ * is one more place, when the handler should be stopped even
+ * before SWIM deletion - quit. Quit deletes the instance only
+ * when all the 'I left' messages are sent, but it happens in a
+ * libev callback in the scheduler fiber where it is impossible
+ * to yield. So to that moment the handler should be dead already.
+ */
+static inline void
+swim_kill_event_handler(struct swim *swim);
```

Comment:
- Let's leave the comments where they are.
- Please remove `inline` - it doesn't make sense to forward declare an inline function.
- The commit message should start with "swim: ...", not with "shutdown: ...". Same comment to commits updating net.box and raft.

Reply:
Fixed.

<details>
<summary>diff</summary>

```diff
diff --git a/src/lib/swim/swim.c b/src/lib/swim/swim.c
index 089505844..ffa32d8d6 100644
--- a/src/lib/swim/swim.c
+++ b/src/lib/swim/swim.c
@@ -548,16 +548,7 @@ struct swim {
        struct trigger on_fiber_shutdown;
 };

-/**
- * Cancel and wait finish of an event handler fiber. That
- * operation is not inlined in the SWIM destructor, because there
- * is one more place, when the handler should be stopped even
- * before SWIM deletion - quit. Quit deletes the instance only
- * when all the 'I left' messages are sent, but it happens in a
- * libev callback in the scheduler fiber where it is impossible
- * to yield. So to that moment the handler should be dead already.
- */
-static inline void
+static void
 swim_kill_event_handler(struct swim *swim);

 /** Put the member into a list of ACK waiters. */
@@ -2223,6 +2214,15 @@ swim_size(const struct swim *swim)
        return mh_size(swim->members);
 }

+/**
+ * Cancel and wait finish of an event handler fiber. That
+ * operation is not inlined in the SWIM destructor, because there
+ * is one more place, when the handler should be stopped even
+ * before SWIM deletion - quit. Quit deletes the instance only
+ * when all the 'I left' messages are sent, but it happens in a
+ * libev callback in the scheduler fiber where it is impossible
+ * to yield. So to that moment the handler should be dead already.
+ */
 static inline void
 swim_kill_event_handler(struct swim *swim)
 {
```

</details>